aws/resource_aws_instance_test.go

```diff
@@ -701,6 +701,44 @@ func TestAccAWSInstance_NetworkInstanceSecurityGroups(t *testing.T) {
 	})
 }
 
+func TestAccAWSInstance_NetworkInstanceRemovingAllSecurityGroups(t *testing.T) {
+	var v ec2.Instance
+
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_instance.foo_instance",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccInstanceNetworkInstanceVPCSecurityGroupIDs(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckInstanceExists(
+						"aws_instance.foo_instance", &v),
+					resource.TestCheckResourceAttr(
+						"aws_instance.foo_instance", "security_groups.#", "0"),
+					resource.TestCheckResourceAttr(
+						"aws_instance.foo_instance", "vpc_security_group_ids.#", "1"),
+				),
+			},
+			{
+				Config: testAccInstanceNetworkInstanceVPCRemoveSecurityGroupIDs(rInt),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
Minor nitpick: `Check` is not required with `ExpectError`, but I'm surprised that `TestStep` didn't complain. 🤷‍♂️ These extra checks can be removed in a followup PR or better yet it won't matter if a `CustomizeDiff` is handling this at plan time as it won't be possible to apply. 👍 

I verified that `TestStep` did the right thing as an error was returned in the acceptance test:

```
2018/03/09 14:51:28 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* aws_instance.foo_instance: VPC-based instances require at least one security group to be attached.
```